### PR TITLE
feat(core,bot-api): Add CompositeContentBuilder

### PR DESCRIPTION
**Pull request by lipis**
## Pull Request Checklist

- [ ] My code is covered by tests
- [ ] I will [merge the PR as breaking change](https://github.com/wireapp/wire-web-packages/wiki/Releases#create-major-release), if the API contract changes

### CODE REVIEW

File: packages/core/src/main/conversation/message/CompositeContentBuilder.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    this.payloadBundle = payloadBundle;
+    this.content = this.payloadBundle.content as CompositeContent;
+  }
+
+  public build(): CompositeMessage | EditedTextMessage {
+    this.payloadBundle.content = this.content;
+    return this.payloadBundle;
+  }
+
+  public withReadConfirmation(expectsReadConfirmation = false): CompositeContentBuilder {
```
Comment (bennyn) [inline packages/core/src/main/conversation/message/CompositeContentBuilder.ts:40]:
```suggestion
  public withReadConfirmation(expectsReadConfirmation: boolean = false): CompositeContentBuilder {
```

File: packages/core/src/main/conversation/message/CompositeContentBuilder.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+export class CompositeContentBuilder {
+  private readonly content: CompositeContent;
+  private readonly payloadBundle: CompositeMessage | EditedTextMessage;
+
+  constructor(payloadBundle: CompositeMessage | EditedTextMessage) {
+    this.payloadBundle = payloadBundle;
+    this.content = this.payloadBundle.content as CompositeContent;
+  }
+
+  public build(): CompositeMessage | EditedTextMessage {
```
Comment (bennyn) [inline packages/core/src/main/conversation/message/CompositeContentBuilder.ts:35]:
```suggestion
  public build(): CompositeMessage {
```

File: packages/core/src/main/conversation/message/CompositeContentBuilder.ts [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    this.payloadBundle.content = this.content;
+    return this.payloadBundle;
+  }
+
+  public withReadConfirmation(expectsReadConfirmation = false): CompositeContentBuilder {
+    if (typeof expectsReadConfirmation !== 'undefined') {
```
Comment (bennyn) [inline packages/core/src/main/conversation/message/CompositeContentBuilder.ts:41]: This `typeof` check can be removed.

File: packages/core/src/main/conversation/message/CompositeContentBuilder.ts [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    if (!this.content.items) {
+      this.content.items = [];
+    }
+    this.content.items = [...this.content.items, Item.create({text})];
+    return this;
+  }
+
+  public addButton(buttonText: string): CompositeContentBuilder {
+    const buttonProtos = [];
+    buttonProtos.push(Button.create({id: '', text: buttonText}));
```
Comment (lipis) [inline packages/core/src/main/conversation/message/CompositeContentBuilder.ts:64]: what should be an `id`

File: packages/core/src/main/conversation/message/CompositeContentBuilder.ts [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+      this.content.legalHoldStatus = legalHoldStatus;
+    }
+    return this;
+  }
+
+  public addText(text: Text): CompositeContentBuilder {
+    if (!this.content.items) {
+      this.content.items = [];
+    }
+    this.content.items = [...this.content.items, Item.create({text})];
```
Comment (bennyn) [inline packages/core/src/main/conversation/message/CompositeContentBuilder.ts:58]:
```suggestion
    this.content.items.push(Item.create({text}));
```

File: packages/core/src/main/conversation/message/CompositeContentBuilder.ts [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+    return this;
+  }
+
+  public addButton(buttonText: string): CompositeContentBuilder {
+    const buttonProtos = [];
+    buttonProtos.push(Button.create({id: '', text: buttonText}));
+
+    if (!this.content.items) {
```
Comment (bennyn) [inline packages/core/src/main/conversation/message/CompositeContentBuilder.ts:66]: Remove these kind of checks and add `this.content.items = []` to the constructor.

File: packages/core/src/main/conversation/message/CompositeContentBuilder.ts [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+      this.content.items = [];
+    }
+    this.content.items = [...this.content.items, ...buttonProtos.map(button => Item.create({button}))];
```
Comment (bennyn) [inline packages/core/src/main/conversation/message/CompositeContentBuilder.ts:69]: Use `Array.push`

File: packages/core/src/main/conversation/message/CompositeContentBuilder.ts [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+
+  public addText(text: Text): CompositeContentBuilder {
+    if (!this.content.items) {
+      this.content.items = [];
+    }
+    this.content.items = [...this.content.items, Item.create({text})];
+    return this;
+  }
+
+  public addButton(buttonText: string): CompositeContentBuilder {
```
Comment (bennyn) [inline packages/core/src/main/conversation/message/CompositeContentBuilder.ts:62]: Remove all `public` declarations since these are the default.

**Pull request merged**